http: require HTTP/0.9 to be explicitly allowed to get accepted

bagder opened the pull request:
CURLOPT_HTTP09 and --http0.9 are added for this purpose.

The default is now kept as before, but added option to explicitly allow it and added a mention to DEPRECATE.md to switch the default to not allowed by support 6 months after the 7.64.0 release.

Fixes #2873

bagder commented:
Oops, it broke a bunch of tests...

bagder commented:
The fuzzer hits the 50 minute limit... :-1:

danielgustafsson commented on docs/DEPRECATE.md:
```diff
@@ -64,3 +64,18 @@ revert if need be.
+### Removal
+
+The support for HTTP/0.9 will be switched to disabled by default in 6 months,
+in the September 2019 release (possilby called curl 7.68.0).
```
s/possilby/possibly/

danielgustafsson commented on docs/cmdline-opts/http0.9.d:
```diff
@@ -0,0 +1,14 @@
+connect with this to non-HTTP servers and still get a response since curl will
+simply transparently downgrade - if allowed.
+
+A future curl version will deny continuing if the resposne isn't at least
```
s/resposne/response/

bagder commented:
Thanks @danielgustafsson, fixed the spelling errors and I've renamed the libcurl option to `CURLOPT_HTTP09_ALLOWED`. Oops, forgot that there were some test failures left to fix...
